tensorboardX/record_writer.py

```diff
@@ -100,8 +100,24 @@ def directory_check(self, path):
         # so we can just skip this check
         pass
 
+    
+class GCSRecordWriterFactory(object):
+    """Factory for event protocol buffer files to GCS."""
 
+    def open(self, path):
+        # Conditional import
+        from tensorflow import gfile
```

Comment:
We should replace this by stealing some code from: https://github.com/tensorflow/tensorboard/blob/84d34b3ec5bd134d28c8dc411df3cee535267610/tensorboard/compat/tensorflow_stub/io/gfile.py#L306 to minimize tensorflow dependency.